cmd/storage-rest-server.go

```diff
@@ -133,47 +133,43 @@ func (s *storageRESTServer) DeleteVolHandler(w http.ResponseWriter, r *http.Requ
 	}
 }
 
-// PrepareFileHandler - fallocate() space for a file.
-func (s *storageRESTServer) PrepareFileHandler(w http.ResponseWriter, r *http.Request) {
+// AppendFileHandler - append to a file.
+func (s *storageRESTServer) AppendFileHandler(w http.ResponseWriter, r *http.Request) {
 	if !s.IsValid(w, r) {
 		return
 	}
 	vars := mux.Vars(r)
 	volume := vars[storageRESTVolume]
 	filePath := vars[storageRESTFilePath]
-	fileSizeStr := vars[storageRESTLength]
-	fileSize, err := strconv.Atoi(fileSizeStr)
+
+	buf := make([]byte, r.ContentLength)
+	_, err := io.ReadFull(r.Body, buf)
 	if err != nil {
 		s.writeErrorResponse(w, err)
 		return
 	}
-	err = s.storage.PrepareFile(volume, filePath, int64(fileSize))
+	err = s.storage.AppendFile(volume, filePath, buf)
 	if err != nil {
 		s.writeErrorResponse(w, err)
 	}
 }
 
-// AppendFileHandler - append to a file.
-func (s *storageRESTServer) AppendFileHandler(w http.ResponseWriter, r *http.Request) {
+// CreateFileHandler - fallocate() space for a file.
+func (s *storageRESTServer) CreateFileHandler(w http.ResponseWriter, r *http.Request) {
 	if !s.IsValid(w, r) {
 		return
 	}
 	vars := mux.Vars(r)
 	volume := vars[storageRESTVolume]
 	filePath := vars[storageRESTFilePath]
 
-	if r.ContentLength < 0 {
-		s.writeErrorResponse(w, errInvalidArgument)
-		return
-	}
-
-	buf := make([]byte, r.ContentLength)
-	_, err := io.ReadFull(r.Body, buf)
+	fileSizeStr := vars[storageRESTLength]
```

Comment:
why this style is preferred? why not let HTTP do the work here i.e r.ContentLength

Reply:
Not a preferred style, I had not copy/pasted it from AppendFile and wrote new code this way. It can be written to use  r.ContentLength